File: web/src/api/cmsCatArt.ts
import { http,resData } from "@/utils/http";
 
// @Tags CmsCatArt
// @Summary 创建CmsCatArt
export const createCmsCatArt = (data?: object) => {
   return http.request<resData>("post", "/cmsCatArt/createCmsCatArt", { data });  
}

// @Tags CmsCatArt
// @Summary 删除CmsCatArt
export const deleteCmsCatArt = (data?: object) => {
   return http.request<resData>("post", "/cmsCatArt/deleteCmsCatArt", { data });  
}
 

// @Tags CmsCatArt
// @Summary 删除ByIdsCmsCatArt
export const deleteCmsCatArtByIds = (data?: object) => {
   return http.request<resData>("post", "/cmsCatArt/deleteCmsCatArtByIds", { data });  
}
 

// @Tags CmsCatArt
// @Summary 更新CmsCatArt
export const updateCmsCatArt = (data?: object) => {
   return http.request<resData>("post", "/cmsCatArt/updateCmsCatArt", { data });  
}

// @Tags CmsCatArt
// @Summary 快速编辑
export const quickEdit = (data?: object) => {
   return http.request<resData>("post", "/cmsCatArt/quickEdit", { data });  
}
 

// @Tags CmsCatArt
// @Summary 用id查询CmsCatArt
export const findCmsCatArt = (params?: object) => {
   return http.request<resData>("get", "/cmsCatArt/findCmsCatArt", { params });  
}
 

// @Tags CmsCatArt
// @Summary 分页获取CmsCatArt列表
export const getCmsCatArtList = (params?: object) => {
   return http.request<resData>("get", "/cmsCatArt/getCmsCatArtList", { params });  
}
 

// @Tags CmsCatArt
// @Summary 导出excelCmsCatArt列表
export const excelList = (params?: object) => {
   return http.request<resData>("get", "/cmsCatArt/excelList", { params });  
}

File: server/router/business/cmsCatArt.go
package business

import (
	"go-cms/api/v1"
	"go-cms/middleware"
	"github.com/gin-gonic/gin"
)

type CmsCatArtRouter struct {
}

// InitCmsCatArtRouter 初始化 CmsCatArt 路由信息
func (s *CmsCatArtRouter) InitCmsCatArtRouter(Router *gin.RouterGroup) {
	router := Router.Group("cmsCatArt").Use(middleware.OperationRecord())
	routerNoRecord := Router.Group("cmsCatArt")
	var apiV1 = v1.ApiGroupApp.BusinessApiGroup.CmsCatArtApi
	{
		router.POST("createCmsCatArt", apiV1.CreateCmsCatArt)   // 新建CmsCatArt
		router.POST("deleteCmsCatArt", apiV1.DeleteCmsCatArt) // 删除CmsCatArt
		router.POST("deleteCmsCatArtByIds", apiV1.DeleteCmsCatArtByIds) // 批量删除CmsCatArt
		router.POST("updateCmsCatArt", apiV1.UpdateCmsCatArt)    // 更新CmsCatArt
	    router.POST("quickEdit", apiV1.QuickEdit)  // 快速编辑	
	}
	{
		routerNoRecord.GET("findCmsCatArt", apiV1.FindCmsCatArt)        // 根据ID获取CmsCatArt
		routerNoRecord.GET("getCmsCatArtList", apiV1.GetCmsCatArtList)  // 获取CmsCatArt列表
        routerNoRecord.GET("excelList", apiV1.ExcelList)  // 分页导出excel CmsCatArt列表
	}
}
